core/dbt/context/base.py

```diff
@@ -457,6 +457,48 @@ def toyaml(
         except (ValueError, yaml.YAMLError):
             return default
 
+    @contextmember("set")
+    @staticmethod
+    def _set(value: Iterable[Any], default: Any = None) -> Optional[Set[Any]]:
+        """The `set` context method can be used to convert any iterable
+        to a sequence of iterable elements that are unique (a set).
+
+        :param value: The iterable
+        :param default: A default value to return if the `value` argument
+            is not an iterable
+
+        Usage:
+            {% set my_list = [1, 2, 2, 3] %}
+            {% set my_set = set(my_list) %}
+            {% do log(my_set) %}  {# {1, 2, 3} #}
+        """
+        try:
+            return set(value)
+        except TypeError:
+            return default
```

Comment:
We've been having some internal discussion around the try/except pattern you see in this file.  We've decided to maintain the current state but also add new partner methods that would allow for failure instead of passing the default back.  Something that would need to be opted in to.  #5167 has some more details of the pattern we would like to follow.

Please add in another function called `try_set` that will fail on a `TypeError`.  Same applies for `zip` below.